pandas/core/arrays/timedeltas.py

```diff
@@ -918,13 +918,9 @@ def sequence_to_td64ns(data, copy=False, unit="ns", errors="raise"):
         copy = copy and not copy_made
 
     elif is_float_dtype(data.dtype):
-        # treat as multiples of the given unit.  If after converting to nanos,
-        #  there are fractional components left, these are truncated
-        #  (i.e. NOT rounded)
-        mask = np.isnan(data)
-        coeff = np.timedelta64(1, unit) / np.timedelta64(1, 'ns')
-        data = (coeff * data).astype(np.int64).view('timedelta64[ns]')
-        data[mask] = iNaT
+        # object_to_td64ns has custom logic for float -> int conversion
+        # to avoid precision issues
```

Comment:
does this have the same perf?

Reply:
No, it is slower. But it is more correct (it is written specifically to handle this case), and it is what was used before 0.24.0 anyway.

I assume we can port the similar logic here to be more performant (to not work element by element), but I would personally leave that for 0.25.0.

Reply:
well the prior fix was for performance and this is a very narrow minor case

so you are causing a rather large perf regression by changing this

Reply:
pls shown asv before / after

Reply:
Converting floats to timedelta is a narrow use case anyway. If you care about performance but not care about precision, you can convert to integers yourself.